### PR TITLE
LM-3187 - write Trades to Azure Tables

### PR DESCRIPTION
**Pull request by nicuceobotaru**

### CODE REVIEW

File: src/Lykke.Job.TradelogBridge/Settings/AppSettings.cs [hunk 1]
```diff
@@ -30,6 +30,9 @@ public class TradelogBridgeSettings
         public int WarningSqlTableSizeInGigabytes { get; set; }
 
         public RabbitMqSettings Rabbit { get; set; }
+
+        [AzureTableCheck]
+        public string TradesConnString { get; set; }
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Job.TradelogBridge/Settings/AppSettings.cs]: It's better to name it `TableStorageConnString`

File: Lykke.Job.TradelogBridge.Azure/TradesRepository.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                var groupedTrades = trades.GroupBy(t => t.PartitionKey);
+                foreach (var group in groupedTrades)
+                {
+                    await _tradesTableStorage.InsertAsync(group);
+                }
+            }
+            catch (Exception ex)
+            {                
+                //don't throw initially, so we don't break existing functionality
+                _log.Error("Error inserting Trades into Table Storage", ex);
```
Comment (KonstantinRyazantsev) [inline Lykke.Job.TradelogBridge.Azure/TradesRepository.cs]: is it ok if some trades will be not inserted in case of transient errors?

Comment (nicuceobotaru) [reply Lykke.Job.TradelogBridge.Azure/TradesRepository.cs]: you're right

**Review by KonstantinRyazantsev — COMMENTED** [review]
Let's implement a separate rabbit mq message subscriber with a separate queue for the new repository to have a more reliable implementation as it was discussed on the call

File: Lykke.Job.TradelogBridge.Azure/TradesRepository.cs [hunk 3]
```diff
@@ -25,18 +26,27 @@ public async Task InsertTradesAsync(List<Trade> trades)
...
+                    await _tradesTableStorage.InsertAsync(trade);
+                }
+                catch (StorageException storageEx)
+                {
+                    if (storageEx.RequestInformation.HttpStatusCode == (int)HttpStatusCode.Conflict)
+                    {
+                        _log.Warning("Duplicated Trade row found", null, trade);
+                        continue;
+                    }
+                    _log.Error("Error inserting Trade into Table Storage", storageEx, storageEx.Message, trade);
```
Comment (KonstantinRyazantsev) [inline Lykke.Job.TradelogBridge.Azure/TradesRepository.cs]: for log.Error it's better to use overload (Exception ex, string message, object context,..), so your invocation should look like:
```
_log.Error(storageEx, "Error inserting Trade into Table Storage", trade);
```

the same for other error messages below. This invocations logs containing class name and method name as `Component` and `Process` log fields respectively, which is common practice in Lykke UK.

File: src/Lykke.Job.TradelogBridge.Services/ReportingTradesSubscriber.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        private async Task ProcessMessageAsync(List<TradeLogItem> arg)
+        {
+            try
+            {
+                var trades = _storageEntityMapper.Map(arg);
+                await _storageTradesRepository.InsertTradesAsync(trades);
+            }
+            catch (Exception ex)
+            {
+                _log.WriteError("ReportingTradesSubscriber.ProcessMessageAsync", arg, ex);
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Job.TradelogBridge.Services/ReportingTradesSubscriber.cs]: the same here and use `Error` not `WriteError`

**Review by KonstantinRyazantsev — APPROVED** [review]

**Pull request merged**